pkg/vkMachinery/csr/generation_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019-2021 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package csr
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"net"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	podNameT string
+	podIP    net.IP
+	csrPEM   []byte
+	keyPEM   []byte
+)
+
+var _ = Describe("", func() {
+	When("Generate a certificate for a virtual node", func() {
+		BeforeEach(func() {
+			podNameT = "podName"
+			podIP = net.ParseIP("10.0.0.1")
+		})
+
+		JustBeforeEach(func() {
+			csrPEM, keyPEM, err = generateVKCertificateBundle(podNameT, podIP)
+		})
+
+		It("Should Create the certificate", func() {
+			Expect(err).NotTo(HaveOccurred())
+			Expect(csrPEM).NotTo(BeNil())
+			Expect(keyPEM).NotTo(BeNil())
+			b, _ := pem.Decode(csrPEM)
+			Expect(b).NotTo(BeNil())
+			csr, err := x509.ParseCertificateRequest(b.Bytes)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(csr.IPAddresses)).To(BeNumerically("==", 1))
+			Expect(len(csr.DNSNames)).To(BeNumerically("==", 1))
+			Expect(csr.IPAddresses[0].String()).To(BeEquivalentTo(podIP.String()))
+			Expect(csr.DNSNames[0]).To(BeEquivalentTo(podNameT))
+			Expect(len(csr.Subject.Organization)).To(BeNumerically("==", 1))
+			Expect(csr.Subject.Organization[0]).To(BeEquivalentTo(csrNodeGroup))
+			Expect(csr.Subject.CommonName).To(BeEquivalentTo(csrNodeGroupMember + podNameT))
+		})
```

Comment:
It is typically better to avoid big It blocks, but split the checks in multiple blocks with the appropriate description. This simplifies troubleshooting in case of failing checks.